src/openrct2/object/TerrainSurfaceObject.cpp

```diff
@@ -74,24 +73,45 @@ void TerrainSurfaceObject::DrawPreview(rct_drawpixelinfo* dpi, int32_t width, in
     }
 }
 
-void TerrainSurfaceObject::ReadJson(IReadObjectContext* context, const json_t* root)
+void TerrainSurfaceObject::ReadJson(IReadObjectContext* context, json_t& root)
 {
-    auto properties = json_object_get(root, "properties");
-    Colour = ObjectJsonHelpers::ParseColour(ObjectJsonHelpers::GetString(properties, "colour"), 255);
-    Rotations = ObjectJsonHelpers::GetInteger(properties, "rotations", 1);
-    Price = ObjectJsonHelpers::GetInteger(properties, "price", 0);
-    Flags = ObjectJsonHelpers::GetFlags<TERRAIN_SURFACE_FLAGS>(
-        properties,
-        { { "smoothWithSelf", TERRAIN_SURFACE_FLAGS::SMOOTH_WITH_SELF },
-          { "smoothWithOther", TERRAIN_SURFACE_FLAGS::SMOOTH_WITH_OTHER },
-          { "canGrow", TERRAIN_SURFACE_FLAGS::CAN_GROW } });
+    auto properties = root["properties"];
 
-    auto jDefault = json_object_get(root, "default");
-    if (json_is_object(jDefault))
+    if (properties.is_object())
     {
-        DefaultEntry = ObjectJsonHelpers::GetInteger(properties, "normal");
-        DefaultGridEntry = ObjectJsonHelpers::GetInteger(properties, "grid");
-        DefaultUndergroundEntry = ObjectJsonHelpers::GetInteger(properties, "underground");
+        Colour = Colour::FromString(Json::GetString(properties["colour"]), 255);
+        Rotations = Json::GetNumber<int8_t>(properties["rotations"], 1);
+        Price = Json::GetNumber<money32>(properties["price"]);
+        Flags = Json::GetFlags<TERRAIN_SURFACE_FLAGS>(
+            properties,
+            { { "smoothWithSelf", TERRAIN_SURFACE_FLAGS::SMOOTH_WITH_SELF },
+              { "smoothWithOther", TERRAIN_SURFACE_FLAGS::SMOOTH_WITH_OTHER },
+              { "canGrow", TERRAIN_SURFACE_FLAGS::CAN_GROW } });
+
+        auto jSpecialArray = properties["special"];
+        if (jSpecialArray.is_array())
+        {
+            for (auto& [i, el] : jSpecialArray.items())
+            {
+                SpecialEntry entry;
+                entry.Index = Json::GetNumber<uint32_t>(el["index"]);
+                entry.Length = Json::GetNumber<int32_t>(el["length"], -1);
+                entry.Rotation = Json::GetNumber<int32_t>(el["rotation"], -1);
+                entry.Variation = Json::GetNumber<int32_t>(el["variation"], -1);
+                entry.Grid = Json::GetBoolean(el["grid"]);
+                entry.Underground = Json::GetBoolean(el["underground"]);
+                SpecialEntries.push_back(std::move(entry));
+            }
+        }
+    }
+
+    auto jDefault = root["default"];
+    if (jDefault.is_object())
+    {
+        // simonj ~ I assume these should be jDefault["normal"] etc.?
```

Comment:
The existing code used properties["normal"] instead of jDefault["normal"]. The defaults aren't actually used as far as I could see in the data files, but I assume that this wasn't intended behaviour.